main/src/ca/uwaterloo/flix/language/phase/Parser.scala

```diff
@@ -489,7 +489,11 @@ class Parser(val source: Source) extends org.parboiled2.Parser {
   object Expressions {
 
     def Block: Rule1[ParsedAst.Expression] = rule {
-      "{" ~ optWS ~ Expression ~ optWS ~ "}" ~ optWS | Assign
+      "{" ~ optWS ~ Expression ~ optWS ~ "}" ~ optWS | Statement
```

Comment:
We should talk about how to properly parse this.

In addition to doing it correctly, we can also think about whether we can add a bit of semi-colon inference.